### PR TITLE
Bug 1768718 - Add new targeting for May 2022 multiplier experiments

### PR DESCRIPTION
**Pull request by punamdahiya**
Need 5 targeting in experiment for series of may multiplier experiments targeting infrequent, casual, regular and core users

https://bugzilla.mozilla.org/show_bug.cgi?id=1768718

### CODE REVIEW

File: app/experimenter/experiments/constants/nimbus.py [hunk 1]
```diff
@@ -505,6 +505,62 @@ class NimbusTargetingConfig:
...
+    name="Regular user (need pin)",
+    slug="regular_user_need_pin",
+    description="Between 14 and 20 days days of activity in the past 28 days needing pin",
+    targeting="{regular_user} && doesAppNeedPin || {sticky}".format(
+        regular_user=TARGETING_REGULAR_USER_URIS.targeting,
+        sticky=TARGETING_STICKY,
+    ),
+    desktop_telemetry="",
+    application_choice_names=(Application.DESKTOP.name,),
+)
```
Comment (Mardak) [inline app/experimenter/experiments/constants/nimbus.py:518]: nit: put each of these next to where other related activity level targeting are (so infrequent then casual then regular then core)

File: app/experimenter/experiments/constants/nimbus.py [hunk 2]
```diff
@@ -505,6 +505,62 @@ class NimbusTargetingConfig:
...
+    targeting="{infrequent_user} && !isDefaultBrowser".format(
+        infrequent_user=TARGETING_INFREQUENT_USER_URIS.targeting,
+    ),
+    desktop_telemetry="",
+    application_choice_names=(Application.DESKTOP.name,),
+)
+
+TARGETING_CASUAL_USER_NEED_DEFAULT = NimbusTargetingConfig(
+    name="Casual user (need default)",
+    slug="Casual_user_need_default",
```
Comment (Mardak) [inline app/experimenter/experiments/constants/nimbus.py:544]: Not sure if slugs need to be lowercase, but others are.

**Review by Mardak — CHANGES_REQUESTED** [review]

File: app/experimenter/experiments/constants/nimbus.py [hunk 3]
```diff
@@ -505,6 +505,62 @@ class NimbusTargetingConfig:
...
+        casual_user=TARGETING_CASUAL_USER_URIS.targeting,
+    ),
+    desktop_telemetry="",
+    application_choice_names=(Application.DESKTOP.name,),
+)
+
+TARGETING_CORE_USER_NEED_DEFAULT = NimbusTargetingConfig(
+    name="Core user (need default)",
+    slug="Core_user_need_default",
+    description="Atleast 21 days days of activity in the past 28 days needing default",
```
Comment (lmegviar) [inline app/experimenter/experiments/constants/nimbus.py:556]: nit: space between 'At' and 'least'

**Comment by Mardak** [comment]
You might need to https://support.circleci.com/hc/en-us/articles/4410418394523-Getting-Started-with-GitHub-SSO for circleci and/or log out/back in before you push updates to get the automation to run.

**Comment by jaredlockhart** [comment]
@punamdahiya Thanks for submitting!  If you address the other feedback in here I can do the automated code formatting and repush and trigger circle.

**Comment by punamdahiya** [comment]
> @punamdahiya Thanks for submitting! If you address the other feedback in here I can do the automated code formatting and repush and trigger circle.
Thanks @jaredlockhart not sure if automation has picked it this time. I have SSH key added to my github may be need an additional setup step

**Comment by punamdahiya** [comment]
@Mardak feedback addressed thanks!

**Review by Mardak — APPROVED** [review]

**Comment by jaredlockhart** [comment]
@punamdahiya Oh perfect, yes CI ran successfully, looks like all the feedback is addressed.  @Mardak has approved, so I'll merge this 👍

**Comment by Mardak** [comment]
So many targeting! 😜
advanced targeting

**Pull request merged**